spyderlib/utils/encoding.py

```diff
@@ -106,7 +106,12 @@ def get_coding(text):
     for line in text.splitlines()[:2]:
         result = CODING_RE.search(to_text_string(line))
         if result:
-            return result.group(1)
+            codec = result.group(1)
+            # sometimes we find a false encoding that can result in errors
+            if codec in CODECS:
+                return codec
+            else:
+                return None

```

Comment:
Actually if the codec is not valid, the loop should continue to the second line. It's a detail but just removing the `else:` block achieves that and simplifies the code.